### PR TITLE
Handle unexpected error response

### PR DESCRIPTION
**Pull request by nasaabg**
This PR fixes an issue described in an original repository:
https://github.com/arsduo/koala/issues/673 - Unfortunately (or not) we use fork :( 

I also lock Faraday to < 2  in order to be able to run `rspec` - it won't work with a higher version (same change was introduced in [original repo](https://github.com/arsduo/koala/commit/021a6aaa89b2853745a33f20181079288e37e518))
And as suggested by @eizengan I also restricted other dependencies for the future.

Additionally few rubocop fixes.

### CODE REVIEW

**Review by cryptomail — APPROVED** [review]
hardens unexpected responses that occur in the wild.
clamps faraday compatibility
incremental improvements and resiliency

File: koala.gemspec [hunk 1]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |gem|
 
   gem.required_ruby_version = '>= 2.1'
 
-  gem.add_runtime_dependency("faraday")
+  gem.add_runtime_dependency("faraday", "< 2")
   gem.add_runtime_dependency("addressable")
```
Comment (eizengan) [inline koala.gemspec]: Unrestricted dependencies are scary! Can we safely pin this as well?

File: lib/koala/api/graph_error_checker.rb [hunk 2]
```diff
@@ -60,11 +60,13 @@ def base_error_info
...
         # hash and fill it with error data.
-        @response_hash ||= begin
-          JSON.parse(body)
-        rescue JSON::ParserError
-          {}
-        end
+        @response_hash ||=
+          begin
+            response_body = JSON.parse(body)
+            response_body.is_a?(Hash) ? response_body : {}
```
Comment (eizengan) [inline lib/koala/api/graph_error_checker.rb:66]: Perfect~

**Review by eizengan — APPROVED** [review]
One optional suggestion. Nice adds, thanks for cleaning things up!

File: koala.gemspec (hunk 1 above)
Comment (nasaabg) [reply koala.gemspec]: You mean for example like that:

`gem.add_runtime_dependency("addressable", ">= 2.8.0")`

I took `>= 2.8.0` from Gemfile.lock

**Review by eizengan — APPROVED** [review]

**Review by edhana — APPROVED** [review]
LGTM

**Pull request merged**